File: 1019-SquaresOfASortedArray/1019-SquaresOfASortedArray.py
# Last updated: 2/9/2026, 9:54:22 PM
class Solution(object):
    def sortedSquares(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        beg = 0
        end = len(nums) - 1
        
        res = [0] * len(nums)
        resPointer = len(nums) - 1
        
        while beg <= end:
            num1 = abs(nums[beg])
            num2 = abs(nums[end])
            
            if num1 >= num2:
                res[resPointer] = num1 * num1
                beg += 1
            else:
                res[resPointer] = num2 * num2
                end -= 1
            
            resPointer -= 1
            
        return res
            
                
        

File: 1019-SquaresOfASortedArray/1019-SquaresOfASortedArray.java
// Last updated: 2/9/2026, 9:54:26 PM
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] returnArray = new int[nums.length];
        int returnPointer = 0;
        int right = 0;
        int left = 0;

        while (nums[right] < 0 && right < nums.length - 1) {
            right++;
        }
        if (nums.length != 1) {
            left = right - 1;
        }

        while (left >= 0 && right < nums.length) {
            if (Math.abs(nums[right]) < Math.abs(nums[left])) {
                returnArray[returnPointer] = Math.abs(nums[right]) * Math.abs(nums[right]);
                right++;
                returnPointer++;
                if (right >= nums.length) break;
            }
            else {
                returnArray[returnPointer] = Math.abs(nums[left]) * Math.abs(nums[left]);
                left--;
                returnPointer++;
                if (left < 0) break;
            }
        }

        while (right < nums.length && returnPointer < nums.length) {
            returnArray[returnPointer] = Math.abs(nums[right]) * Math.abs(nums[right]);
            right++;
            returnPointer++;
        }

        while (left >= 0 && returnPointer < nums.length) {
            returnArray[returnPointer] = Math.abs(nums[left]) * Math.abs(nums[left]);
            left--;
            returnPointer++;
        }


        return returnArray;
    }
}